server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/shards/TransportCatShardsActionIT.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.shards;
+
+import org.opensearch.action.admin.cluster.state.ClusterStateResponse;
+import org.opensearch.action.admin.indices.stats.IndicesStatsResponse;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.Strings;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.List;
+
+import static org.opensearch.cluster.routing.UnassignedInfo.INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING;
+import static org.opensearch.search.SearchService.NO_TIMEOUT;
+
+@OpenSearchIntegTestCase.ClusterScope(numDataNodes = 0, scope = OpenSearchIntegTestCase.Scope.TEST)
+public class TransportCatShardsActionIT extends OpenSearchIntegTestCase {
+
+    public void testCatShardsWithSuccessResponse() {
+        internalCluster().startClusterManagerOnlyNodes(1);
+        List<String> nodes = internalCluster().startDataOnlyNodes(3);
+        createIndex(
+            "test",
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 2)
+                .put(INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.getKey(), "60m")
+                .build()
+        );
+        ensureGreen("test");
+
+        final CatShardsRequest shardsRequest = new CatShardsRequest();
+        shardsRequest.setCancelAfterTimeInterval(NO_TIMEOUT);
+        shardsRequest.setIndices(Strings.EMPTY_ARRAY);
+        client().execute(CatShardsAction.INSTANCE, shardsRequest, new ActionListener<CatShardsResponse>() {
+            @Override
+            public void onResponse(CatShardsResponse catShardsResponse) {
+                ClusterStateResponse clusterStateResponse = catShardsResponse.getClusterStateResponse();
+                IndicesStatsResponse indicesStatsResponse = catShardsResponse.getIndicesStatsResponse();
+                for (ShardRouting shard : clusterStateResponse.getState().routingTable().allShards()) {
+                    assertEquals("test", shard.getIndexName());
+                    assertNotNull(indicesStatsResponse.asMap().get(shard));
+                }
+            }
+
+            @Override
+            public void onFailure(Exception e) {}
+        });
+    }
```

Comment:
Tried two ways to simulate cancellation timeout in ITs - 
* Tried dropping master node so that cluster.state will take sometime also scheduled a new thread will bring back the master node and meanwhile cancellation timeout will be expired then when the next indices.stats call happens it will fail. It's not working properly, need more time for investigation.
* Checked Search cancellation, there to delay the search action they have blocked the plugin of shards and eventually fail the action.

Reply:
Added IT for cancellation timeout expiry.